http-client/src/main/java/io/avaje/http/client/HttpApiProvider.java

```diff
@@ -1,20 +1,20 @@
 package io.avaje.http.client;
 
+import java.util.Map;
+
 /**
  * Provides http client implementations for an interface.
  *
  * @param <T> The interface type
  */
+@FunctionalInterface
 public interface HttpApiProvider<T> {
 
-  /**
-   * Return the interface type this API implements.
-   */
-  Class<T> type();
+  /** Return the interface type this API implements. */
+  default Class<T> type() {
+    throw new UnsupportedOperationException();
```

Comment:
This does not look right to me ... hmm

Reply:
basically, type should not be used

Reply:
I'm keeping it only for the sake of backwards compatibility

Reply:
when 2.x rolls around I want to take it out actually